lynx/src/main/java/com/github/pedrovgs/lynx/LynxConfig.java

```diff
@@ -55,10 +58,21 @@ public LynxConfig setMaxNumberOfTracesToShow(int maxNumberOfTracesToShow) {
   }
 
   public LynxConfig setFilter(String filter) {
+    if (filter == null) {
+      throw new NullPointerException("filter can't be null");
```

Comment:
I was using IllegalArgumentException, but Effective Java Item 60: "Arguably, all erroneous method invocations boil down to an illegal argument or illegal state, but other exceptions are standardly used for certain kinds of illegal arguments and states. If a caller passes null in some parameter for which null values are prohibited, convention dictates that NullPointerException be thrown rather than IllegalArgumentException.

Bloch, Joshua (2008-05-08). Effective Java (2nd Edition) (Kindle Locations 5179-5182). Pearson Education (USA). Kindle Edition. "
recommends to use in this case NPE and Java 7 API too: http://docs.oracle.com/javase/7/docs/api/java/util/Objects.html#requireNonNull(T)

For consistency with the repo I will use IAE, though